### PR TITLE
fix: distinguish property cache

### PR DESCRIPTION
**Pull request by popomore**
##### Checklist

- [ ] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)

ContextLoader

### CODE REVIEW

File: lib/loader/context_loader.js [hunk 1]
```diff
@@ -44,13 +44,21 @@ class ContextLoader extends FileLoader {
...
+    const classLoader = this[CLASSLOADER];
+    const property = options.property;
 
-    Object.defineProperty(app.context, options.property, {
+    // define ctx.service
+    Object.defineProperty(app.context, property, {
       get() {
-        if (!this[classLoader]) {
-          this[classLoader] = getInstance(target, this);
+        if (!classLoader.has(property)) {
```
Comment (fengmk2) [inline lib/loader/context_loader.js]: 这样写可以减少一次 Map 寻址

```js
let instance = classLoader.has(property);
if (!instance) {
  instance = getInstance(target, this);
  classLoader.set(property, instance);
}
return instance;
```

**Comment by popomore** [comment]
改了

**Comment by fengmk2** [comment]
+1

**Comment by fengmk2** [comment]
1.3.2

**Pull request merged**